### PR TITLE
EWS: Raise 404 instead of 500 when location is archived or doesn't exist

### PR DESCRIPTION
**Pull request by kkrampa**
http://manage.dimagi.com/default.asp?186518#1041175

@gcapalbo 
I guess that some users are getting email reports for archived locations (I will make fix for this in different PR).

### CODE REVIEW

File: custom/ewsghana/reports/__init__.py [hunk 1]
```diff
@@ -318,15 +314,7 @@ def get_report_context(self, data_provider):
...
-        if not self.report_config.get('location_id'):
-            return False
-        sql_location = SQLLocation.objects.get(location_id=self.report_config['location_id'], is_archived=False)
-        reporting_types = [
-            location_type.name
-            for location_type in Domain.get_by_name(self.domain).location_types
-            if not location_type.administrative
-        ]
-        return sql_location.location_type.name in reporting_types
+        return not self.location.location_type.administrative
```
Comment (gcapalbo) [inline custom/ewsghana/reports/__init__.py]: nice, this looks a lot cleaner

**Comment by gcapalbo** [comment]
@kkrampa can we exclude archived locations from the filters, or is that already happening? and if we're already doing that, how are people running the report against an archived location?

**Comment by kkrampa** [comment]
Archived locations are excluded from filters. As I wrote someone have to receive email reports for archived location.

**Comment by gcapalbo** [comment]
ok, is there a way to stop sending email reports for archived locations, or is that the fix you mentioned you'd put in a separate pr?

**Comment by kkrampa** [comment]
@gcapalbo #8983

**Pull request merged**